trace-utils/src/trace_utils.rs

```diff
@@ -31,63 +33,21 @@ macro_rules! parse_string_header {
     }
 }
 
-#[derive(Debug, Deserialize, Serialize, Clone, PartialEq)]
-pub struct Span {
-    service: Option<String>,
-    name: String,
-    resource: String,
-    trace_id: u64,
-    span_id: u64,
-    parent_id: Option<u64>,
-    start: i64,
-    duration: i64,
-    error: Option<i32>,
-    meta: HashMap<String, String>,
-    metrics: Option<HashMap<String, f64>>,
-}
-
 pub async fn get_traces_from_request_body(body: Body) -> anyhow::Result<Vec<Vec<pb::Span>>> {
     let buffer = hyper::body::aggregate(body).await.unwrap();
 
-    let traces: Vec<Vec<Span>> = match rmp_serde::from_read(buffer.reader()) {
+    let traces: Vec<Vec<pb::Span>> = match rmp_serde::from_read(buffer.reader()) {
         Ok(res) => res,
         Err(err) => {
             anyhow::bail!("Error deserializing trace from request body: {}", err)
```

Comment:
```suggestion
            anyhow::bail!("Error deserializing trace from request body: {err}")
```

You will probably be required to fix these if you upgrade Rust because I think clippy started complaining about that in 1.64.0.